SwanCustomEnvironments/swancustomenvironments/templates/customenvs.html

```diff
@@ -0,0 +1,160 @@
+    position: absolute;
+    left: 50%;
+    transform: translateX(-50%);
+    color: white;
+    border: none;
+    border-radius: 5px;
+    padding: 10px 20px;
+    font-size: 16px;
+    cursor: pointer;
+    transition: background-color 0.3s ease;
+}
+#next_button {
+    display: none;
+    text-align: center;
+    background-color: #0454a4;
+    position: absolute;
+    left: 50%;
+    transform: translateX(-50%);
+    color: white;
+    border: none;
+    border-radius: 5px;
+    padding: 10px 20px;
+    font-size: 16px;
+    cursor: pointer;
+    transition: background-color 0.3s ease;
+}
+</style>
+{% endblock %}
+
+{% block params %}
+{{super()}}
+data-base-url="{{base_url | urlencode}}"
+{% endblock %}
+
+{% block site %}
+
+<div class="container">
+    <div class="row">
+        <div class="swan-info">
+            <div id="swan-loader">
+                <div class="loader-circle">
+                    <img src="{{hub_prefix}}static/swan/logos/{{swan_logo_filename}}">
+                </div>
+                <div class="loader-line-mask">
+                    <div class="loader-line"></div>
+                </div>
+                <span class="text">Creating environment...</span>
+            </div>
+        </div>
+        <br><br>
+        <div id="output-box"></div>
+        <br>
+        <button id="back_button">Back</button>
+        <button id="next_button">Next</button>
+    </div>
+</div>
+
+{% endblock %}
+
+{% block script %}
+    {{super()}}
+
+<script type = "text/javascript" >
+    
+    var base_url = document.body.getAttribute('data-base-url');
+    
+    const urlParams = new URLSearchParams(window.location.search);
+    var env_name = urlParams.get('env') || "";
+    var repository = urlParams.get('repo') || "";
+    var accpy_version = urlParams.get('accpy') || "";
+    
+    var project_folder = '/';
+    if (repository.startsWith('http')) {
+        // Extract repository name from URL
+        const pattern = /^https?:\/\/[^/]+\/([^/\s]+)\/([^/\s]+)/;
+        const match = repository.match(pattern);
+        project_folder = match ? `/tree/SWAN_projects/${match[2]}` : '';
+    } else {
+        // Extract CERNBox path for EOS path repository, e.g /eos/user/u/username/project/local -> /tree/project/local
+        const pattern = /^\/eos\/user\/([^/]+)\/([^/]+)(\/.*)?$/;
+        const match = repository.match(pattern);
+        project_folder = match && match[3] ? `/tree/${match[3]}` : '/'
+    }
+
+    var outputBox = document.getElementById('output-box');
+    var loader = document.getElementById('swan-loader');
+    var back_button = document.getElementById('back_button');
+    var next_button = document.getElementById('next_button');
+    var has_failed = false;
+    var eventSource = new EventSource(`${base_url}api/customenvs?env=${encodeURIComponent(env_name)}&repo=${encodeURIComponent(repository)}&accpy=${encodeURIComponent(accpy_version)}`);
+    
+    eventSource.onmessage = (event) => {
```

Comment:
Some comment before this line? Do not be shy on commenting on the key parts of your code.